wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneResolver.swift

```diff
@@ -92,65 +92,98 @@ public final class OneOnOneResolver: OneOnOneResolverInterface {
-                    return
-                }
+            return await resolveCommonUserProtocolNone(with: userID, in: context)
+        case .mls:
+            return try await resolveCommonUserProtocolMLS(with: userID, in: context)
+        case .proteus:
+            return resolveCommonUserProtocolProteus()
+        case .mixed:
+            return resolveCommonUserProtocolMixed()
+        }
+    }
 
-                let selfUser = ZMUser.selfUser(in: context)
+    private func resolveCommonUserProtocolNone(
+        with userID: QualifiedID,
+        in context: NSManagedObjectContext
+    ) async -> OneOnOneConversationResolution {
+        WireLogger.conversation.debug("no common protocols found")
+
+        await context.perform {
+            guard
+                let otherUser = ZMUser.fetch(with: userID, in: context),
+                let conversation = otherUser.oneOnOneConversation
+            else {
+                return
+            }
 
-                // The reason we need to add this first if check here is so to avoid running the same code path again
-                // if conversation has been marked as read only before.
-                if !conversation.isForcedReadOnly {
-                    if !selfUser.supportedProtocols.contains(.mls) {
-                        conversation.appendMLSMigrationMLSNotSupportedForSelfUser(user: selfUser)
-                    } else if !otherUser.supportedProtocols.contains(.mls) {
-                        conversation.appendMLSMigrationMLSNotSupportedForOtherUser(user: otherUser)
-                    }
+            let selfUser = ZMUser.selfUser(in: context)
+
+            // The reason we need to add this first if check here is so to avoid running the same code path again
+            // if conversation has been marked as read only before.
+            if !conversation.isForcedReadOnly {
+                if !selfUser.supportedProtocols.contains(.mls) {
+                    conversation.appendMLSMigrationMLSNotSupportedForSelfUser(user: selfUser)
+                } else if !otherUser.supportedProtocols.contains(.mls) {
+                    conversation.appendMLSMigrationMLSNotSupportedForOtherUser(user: otherUser)
                 }
+            }
 
-                conversation.isForcedReadOnly = true
+            conversation.isForcedReadOnly = true
+        }
 
-            }
-            return .archivedAsReadOnly
+        return .archivedAsReadOnly
+    }
 
-        case .mls:
-            WireLogger.conversation.debug("should resolve to mls 1-1 conversation")
+    private func resolveCommonUserProtocolMLS(
+        with userID: QualifiedID,
+        in context: NSManagedObjectContext
+    ) async throws -> OneOnOneConversationResolution {
+        WireLogger.conversation.debug("should resolve to mls 1-1 conversation")
 
-            guard let migrator else {
-                throw OneOnOneResolverError.migratorNotFound
-            }
+        guard let migrator else {
```

Comment:
Could the `migrator` be made non-optional?

Reply:
@agisilaos please help me here :)
I had the same question when it became optional, but remember that there was case where the migrator was not initialised but the resolver should work for the other cases (proteus, mixed)... so the answer was no, it needs to be optional.

Reply:
It was in #924 and I think we can not change it for now, unfortunately.